### DISCUSSION TITLE
Problem fitting a RSF model

### DISCUSSION THREAD

**Discussion by tomas-silveira** [sebp/scikit-survival#263] [Q&A]
Hello, I'm trying to fit a RandomForestModel using the hyperparameters from the tuturial on my dataset:

```
rsf = RandomSurvivalForest(n_estimators=1000,
                           min_samples_split=10,
                           min_samples_leaf=15,
                           max_features="sqrt",
                           n_jobs=-1,
                           random_state=SEED)
```

However, I'm getting the following error after trying to instantiate the rsf object:
`Can't instantiate abstract class RandomSurvivalForest with abstract methods _set_oob_score`

Do you know what's causing this? I dont know if this relevant, but the dataset used has around 130 features and several hundred thousand instances. Ive used the `get_x_y` method to convert my dataset to the required format.
Thank you!

**Comment by sebp** [comment]
Maybe your are using the wrong version of sklearn.

Can you please post the output of

import sklearn; sklearn.show_versions()
import sksurv; print("sksurv:", sksurv.__version__)

**Reply by tomas-silveira** [reply]
Hello sebp, here goes the output:

```
System:
    python: 3.7.7 (default, Mar 26 2020, 15:48:22)  [GCC 7.3.0]
executable: /***********/bin/python
   machine: Linux-3.10.0-1160.15.2.el7.x86_64-x86_64-with-centos-7.8.2003-Core

Python dependencies:
          pip: 20.2.1
   setuptools: 49.3.1.post20200810
      sklearn: 0.23.2
        numpy: 1.19.1
        scipy: 1.4.1
       Cython: 0.29.21
       pandas: 1.1.0
   matplotlib: 3.2.2
       joblib: 0.14.0
threadpoolctl: 2.1.0

Built with OpenMP: True
sksurv: 0.17.1
```

**Comment by sebp** [comment]
You need sklearn 1.0, but you have version 0.23.2.

Running `pip check` should tell you that something is wrong.

**Comment by tomas-silveira** [comment]
This is the output i get for the pip check:
```
pip check
spyder 4.1.4 requires pyqt5, which is not installed.
spyder 4.1.4 requires pyqtwebengine, which is not installed.
qdarkstyle 2.8.1 requires helpdev, which is not installed.
pytest-astropy 0.8.0 requires pytest-cov, which is not installed.
pytest-astropy 0.8.0 requires pytest-filter-subpackage, which is not installed.
tensorflow 2.3.0 has requirement numpy<1.19.0,>=1.16.0, but you have numpy 1.19.1.
```

No mention about any incompability with your package... Also, Its strange because my scikit-learn version is 1.0.2...
```
python -m pip show scikit-learn
Name: scikit-learn
Version: 1.0.2
Summary: A set of python modules for machine learning and data mining
Home-page: http://scikit-learn.org
Author: None
Author-email: None
License: new BSD
Location: /home/tom_ts/.local/lib/python3.7/site-packages
Requires: numpy, joblib, threadpoolctl, scipy
Required-by: scikit-survival, pysurvival, imbalanced-learn, eli5, sklearn, shap, scikit-uplift, scikit-optimize, pylift, lightgbm, causalml, causallift, category-encoders
```

**Reply by sebp** [reply]
I don't know why pip and sklearn would report different versions. Maybe you have two different versions installed, one in `~/.local/lib` and the other in `/usr/lib`? Create a new virtual environment might be the best option.

**Reply by tomas-silveira** [reply]
Managed to get it to work!
